### ISSUE TITLE
feat: Sampler — append external data to train split

### ISSUE THREAD

**Issue by sunkusun9** [sunkusun9/ml-labs#95]
## Background

In the kaggle s6e3 experiment, the original dataset (before synthetic augmentation) is available separately. Mixing the original data into training improved model performance. To systematize this kind of experiment, a **Sampler** abstraction is needed.

### Base class: `Sampler`

```python
class Sampler:
    def sample(self, train_data, split_idx):
        raise NotImplementedError
```

- Called per split, receives the current split's train `DataWrapper`
- Returns augmented train data (row-direction)
- Lives in a new `mllabs/sampler/` subpackage (or `_sampler.py`)

### `AppendSampler`

```python
class AppendSampler(Sampler):
    def __init__(self, data, v=None):
        ...
    def sample(self, train_data, split_idx):
        # row-concat: train_data || self.data (optionally column-filtered by v)
        ...
```

- `data`: external dataset to append (DataWrapper or raw DataFrame)
- `v`: optional column selector (same interface as `var_spec`)
- Result: `concat([train_data, extra], axis=0)`

## Integration point

`Trainer.get_node_output()` — when `node is None` (DataSource), after slicing `train_idx`:

```python
train_data = self.data.iloc(train_idx)
# ↓ apply sampler here
if self.sampler is not None:
    train_data = self.sampler.sample(train_data, split_idx)
```

`Trainer.__init__` gets a `sampler=None` parameter, saved/loaded alongside splitter.

## Use case (s6e3)

```python
from mllabs.sampler import AppendSampler

original = DataWrapper.wrap(original_df)
trainer = exp.get_trainer("skf5")
trainer = exp.make_trainer(..., sampler=AppendSampler(original))
```

**Comment by sunkusun9** [comment]
## Implementation Note (differs from original design)

The original design proposed a `Sampler` abstraction integrated into `Trainer.get_node_output()`. The actual implementation went a different direction:

### Appending external data → `Experimenter.aug_data`

Instead of `AppendSampler`, the `Experimenter` (and `create`/`load`) now accepts an `aug_data` parameter. When set, it is row-concatenated into the inner train split inside `get_node_output(node=None)`. No Trainer-level changes needed.

```python
exp = Experimenter.load(path, data=df, aug_data=original_df)
```

### Resampling at fit time → `mllab_sampler` in node params

A `Sampler` base class lives in `mllabs.sampler`. It receives `fit_params` (the native dict produced by the adapter's `get_fit_params`) and returns a modified version. Set it via `set_node(..., params={'mllab_sampler': ...})`. The processor extracts and applies it before calling `obj.fit()` / `obj.fit_transform()`, never passing it to the underlying estimator.

```python
from mllabs.sampler import ImbLearnSampler
from imblearn.over_sampling import SMOTE

exp.set_node('lgb', ..., params={'mllab_sampler': ImbLearnSampler(SMOTE()), ...})
```
